src/headset/headset.h

```diff
@@ -56,6 +58,48 @@ typedef struct {
 typedef vec_t(Controller*) vec_controller_t;
 typedef void (*headsetRenderCallback)(HeadsetEye eye, void* userdata);
 
+
+typedef struct  {
+    int (*isAvailable)();
+    void (*init)();
+    void (*destroy)();
+    void (*poll)();
+    int (*isPresent)();
+    HeadsetType (*getType)();
+    HeadsetOrigin (*getOriginType)();
+    int (*isMirrored)();
+    void (*setMirrored)(int mirror);
+    void (*getDisplayDimensions)(int* width, int* height);
+    void (*getClipDistance)(float* clipNear, float* clipFar);
+    void (*setClipDistance)(float clipNear, float clipFar);
+    float (*getBoundsWidth)();
+    float (*getBoundsDepth)();
+    void (*getBoundsGeometry)(float* geometry);
+    void (*getPosition)(float* x, float* y, float* z);
+    void (*getEyePosition)(HeadsetEye eye, float* x, float* y, float* z);
+    void (*getOrientation)(float* angle, float* x, float* y, float* z);
+    void (*getVelocity)(float* x, float* y, float* z);
+    void (*getAngularVelocity)(float* x, float* y, float* z);
+    vec_controller_t* (*getControllers)();
+    int (*controllerIsPresent)(Controller* controller);
+    ControllerHand (*controllerGetHand)(Controller* controller);
+    void (*controllerGetPosition)(Controller* controller, float* x, float* y, float* z);
+    void (*controllerGetOrientation)(Controller* controller, float* angle, float* x, float* y, float* z);
+    float (*controllerGetAxis)(Controller* controller, ControllerAxis axis);
+    int (*controllerIsDown)(Controller* controller, ControllerButton button);
+    int (*controllerIsTouched)(Controller* controller, ControllerButton button);
+    void (*controllerVibrate)(Controller* controller, float duration, float power);
+    ModelData* (*controllerNewModelData)(Controller* controller);
+    void (*renderTo)(headsetRenderCallback callback, void* userdata);
+    void (*update)(float dt);
+} HeadsetImpl;
```

Comment:
Do you mind renaming this to something without an abbreviation?  Maybe `HeadsetInterface` or something.